### PR TITLE
Refactor code + Add CORS header

### PR DESCRIPTION
**Pull request by nlegoff**

### CODE REVIEW

**Comment by romainneutron** [comment]
Could you have a look at travis build ?

**Comment by nlegoff** [comment]
@romainneutron can you review this PR ?

**Comment by nlegoff** [comment]
@romainneutron PR is cleaned up

File: .gitignore [hunk 1]
```diff
@@ -1 +1,5 @@
-nbproject
+/nbproject
```
Comment (romainneutron) [inline .gitignore]: please remove nbproject

**Comment by romainneutron** [comment]
Please fix the tests !

**Comment by romainneutron** [comment]
Please fix the tests

**Comment by nlegoff** [comment]
@romainneutron Tests fixed, the only one which is faling is the one concerning the distance sorting

**Pull request merged**